feat(core): add `invalidateOnUnmount` and `invalidateOnClose` prop on `autoSave` feature (#4903)
refinedev/refine: .changeset/fast-dodos-think.md

---
"@refinedev/core": patch
---

> feat: `autoSave` mutation not invalide when it mutation successful. Now `list`,`many` and `detail` queries will be invalidated by default. Also added [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to work when `unmount`. However, you can use the `invalidates` prop if you want to customize it.

Reviewer: ```suggestion
feat: `autoSave` mutation wasn't being invalidated when the mutation is successful. Now `list`,`many` and `detail` queries will be invalidated by default. Also added [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to to trigger invalidate on `unmount`. You can use `invalidates` prop to customize it.
```

---
"@refinedev/core": patch
---

feat: add [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to [`useForm`](https://refine.dev/docs/api-reference/core/hooks/useForm) hook.

> From now, you can use the [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to invalidate queries upon unmount.

Reviewer: ```suggestion
From now on, you can use the [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to invalidate queries upon unmount.
```
